File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.ArrayList;

class Solution {
	public int[] solution(int n, long left, long right) {
		ArrayList<Integer> list = new ArrayList<Integer>();
		
		for(long i = left; i <= right; i++) {
			long row = i % n + 1;
			long col = i / n + 1;
			int num = (int) (row > col ? row : col);
			list.add(num);
		}

		return list.stream().mapToInt(Integer::intValue).toArray();
	}
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.py
def solution(n, left, right):
    answer = []

    for i in range(left, right + 1):
        num = 1

        # n 을 기준으로 칸을 나눴을 때의 인덱스
        row = i // n 
        col = i % n

        if row > col:
            num += i // n
        else:
            num = col + 1

        answer.append(num)    

    return answer
